### PR TITLE
Add checks for file existence

### PR DESCRIPTION
**Pull request by alexjhawk**

### CODE REVIEW

**Review by oliver-walker-hms — DISMISSED** [review]

File: src/com/hms_networks/americas/sc/fileutils/FileAccessManager.java [hunk 1]
```diff
@@ -60,6 +65,11 @@ public static String readFileToString(String filename) throws IOException {
    * @throws IOException if unable to access or write file
    */
   public static void writeStringToFile(String filename, String contents) throws IOException {
+    // Verify folders exist and file exists
+    File file = new File(filename);
+    file.mkdirs();
+    if (!file.exists()) file.createNewFile();
```
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/fileutils/FileAccessManager.java]:
```suggestion
    if (!file.exists()) {
      file.createNewFile();
    }
```

File: src/com/hms_networks/americas/sc/fileutils/FileAccessManager.java [hunk 2]
```diff
@@ -25,6 +26,10 @@
    * @throws IOException if unable to access or read file
    */
   public static String readFileToString(String filename) throws IOException {
+    // If file does not exist, return null
+    File file = new File(filename);
+    if (!file.exists()) return null;
```
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/fileutils/FileAccessManager.java]: best to avoid multiple returns.

**Review by TomKimsey — CHANGES_REQUESTED** [review]

File: src/com/hms_networks/americas/sc/fileutils/FileAccessManager.java (hunk 2 above)
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/fileutils/FileAccessManager.java]: refactored

File: src/com/hms_networks/americas/sc/fileutils/FileAccessManager.java (hunk 1 above)
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/fileutils/FileAccessManager.java]: done

File: src/com/hms_networks/americas/sc/fileutils/FileAccessManager.java [hunk 3]
```diff
@@ -25,31 +26,41 @@
    * @throws IOException if unable to access or read file
    */
   public static String readFileToString(String filename) throws IOException {
-    // Create input stream and buffered reader
-    InputStream inputStream = new FileInputStream(filename);
-    BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
+    // If file does not exist, return null
+    File file = new File(filename);
+    String returnVal;
```
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/fileutils/FileAccessManager.java]: if you set returnVal to null here you dont need the else

**Review by TomKimsey — DISMISSED** [review]
approved but made a note about the the else statement

File: src/com/hms_networks/americas/sc/fileutils/FileAccessManager.java (hunk 3 above)
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/fileutils/FileAccessManager.java]: changed

**Review by oliver-walker-hms — APPROVED** [review]

**Review by TomKimsey — APPROVED** [review]

**Pull request merged**